### PR TITLE
Bdog 914

### PR DESCRIPTION
**Pull request by christopherjturner**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: python
 python:
-  - "2.7"
+  - "3.7"
```
Comment (cjwebb) [inline .travis.yml:3]: Should we also run against Python 3.8? Have a matrix of supported Python versions?
Or does this just setup the tox script, in which case we should change that to run against 3.8 too?

File: servicemanager/smdownload.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import requests
+import time
+import sys
+
+
+def _report_hook(count, block_size, total_size):
+    global start_time
+    global last_update
```
Comment (cjwebb) [inline servicemanager/smdownload.py:8]: Why globals?
I wonder, as part of me would like to try parallelizing SM downloads at some point in the future.

File: test/it/testbase.py [hunk 3]
```diff
@@ -38,12 +40,41 @@ def set_up_and_clean_workspace(self):
...
+    def setup_local_git(self):
+        workspace_dir = os.path.join(os.path.dirname(__file__), "workspace")
+        testapp_dir = os.path.join(os.path.dirname(__file__), "../testapps/basicplayapp")
+        os.makedirs(os.path.join(workspace_dir,  "git"))
+        shutil.copytree(testapp_dir, os.path.join(workspace_dir,  "git", "basicplayapp"))
+        os.chdir(os.path.join(workspace_dir, "git", "basicplayapp"))
+        command = "git init && git add . && git commit -m 'test'"
+        ps_command = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, universal_newlines=True)
+        ps_command.communicate()
+        os.chdir(workspace_dir)
```
Comment (cjwebb) [inline test/it/testbase.py:52]: Wondered if it was possible for this to fail, somehow? Is the teardown guaranteed to always run?

Comment (christopherjturner) [reply test/it/testbase.py:52]: set_up_and_clean_workspace will delete the whole workspace folder when the test runs so shouldn't be an issue if teardown fails

File: servicemanager/smdownload.py (hunk 2 above)
Comment (christopherjturner) [reply servicemanager/smdownload.py:8]: this was a cut/paste of the original progress meter. should be easy enough to refactor the globals out of it.
parallelizing the downloads would be cool, though rendering the progress might get a bit more complex

**Pull request merged**